src/Detector/Framework.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Utopia\Detector\Detector;
+
+use Utopia\Detector\Detection\Framework as FrameworkDetection;
+use Utopia\Detector\Detector;
+
+class Framework extends Detector
+{
+    public function __construct(
+        protected array $inputs,
+        protected string $packager = 'npm'
+    ) {}
+
+    public function detect(): ?FrameworkDetection
+    {
+        foreach ($this->options as $detector) {
+            $detectorFiles = $detector->getFiles();
+
+            foreach ($this->inputs as $input) {
+                foreach ($detectorFiles as $pattern) {
```

Comment:
Do we have a strong reason for why framework does exact array match, but package manager detection only does `array_intersect` expecting at least 1 match?